lib/ansible/modules/cloud/cloudstack/cs_role_permission.py

```diff
@@ -0,0 +1,307 @@
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: cs_role_permission
+short_description: Manages role permissions on Apache CloudStack based clouds.
+description:
+    - Create, update and remove CloudStack role permissions.
+version_added: '2.6'
+author: "David Passante (@dpassante)"
+options:
+  name:
+    description:
+      - The API name of the permission.
+    required: true
+  role:
+    description:
+      - Name or ID of the role.
+    required: true
+  permission:
+    description:
+      - The rule permission, allow or deny. Defaulted to deny.
+    choices: [ allow, deny ]
+    required: false
+    default: deny
+  state:
+    description:
+      - State of the role permission.
+    choices: [ present, absent ]
+    required: false
+    default: present
+  description:
+    description:
+      - The description of the role permission.
+    required: false
+  parent:
+    description:
+      - The parent role permission uuid. use 0 to move this rule at the top of the list.
+    required: false
+extends_documentation_fragment: cloudstack
+'''
+
+EXAMPLES = '''
+# Create a role permission
+- local_action:
+    module: cs_role_permission
+    role: "My_Custom_role"
+    name: "createVPC"
+    permission: "allow"
+    description: "My comments"
+
+# Remove a role permission
+- local_action:
+    module: cs_role_permission
+    state: absent
+    role: "My_Custom_role"
+    name: "createVPC"
+
+# Updete a system role permission
+- local_action:
+    module: cs_role_permission
+    role: "Domain Admin"
+    name: "createVPC"
+    permission: "deny"
+
+# Updade rules order. move the rule at the top of list
+- local_action:
+    module: cs_role_permission
+    role: "Domain Admin"
+    name: "createVPC"
+    parent: 0
+'''
+
+RETURN = '''
+---
+id:
+  description: The ID of the role permission.
+  returned: success
+  type: string
+  sample: a6f7a5fc-43f8-11e5-a151-feff819cdc9f
+name:
+  description: The API name of the permission.
+  returned: success
+  type: string
+  sample: createVPC
+permission:
+  description: The permission type of the api name.
+  returned: success
+  type: string
+  sample: allow
+roleid:
+  description: The ID of the role to which the role permission belongs.
+  returned: success
+  type: string
+  sample: c6f7a5fc-43f8-11e5-a151-feff819cdc7f
+description:
+  description: The description of the role permission
+  returned: success
+  type: string
+  sample: Deny createVPC for users
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.cloudstack import (
+    AnsibleCloudStack,
+    cs_argument_spec,
+    cs_required_together,
+)
+
+
+class AnsibleCloudStackRolePermission(AnsibleCloudStack):
+
+    def __init__(self, module):
+        super(AnsibleCloudStackRolePermission, self).__init__(module)
+        self.returns = {
+            'id': 'id',
+            'roleid': 'roleid',
```

Comment:
i would prefer `role_id` here

Reply:
would you mind change this to the following?
~~~
'roleid': 'role_id',
~~~

Reply:
Sorry! Done.